server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -1061,6 +1062,9 @@ public Node start() throws NodeValidationException {
         TransportService transportService = injector.getInstance(TransportService.class);
         transportService.getTaskManager().setTaskResultsService(injector.getInstance(TaskResultsService.class));
         transportService.getTaskManager().setTaskCancellationService(new TaskCancellationService(transportService));
+
+        TaskAwareRunnable.setListener(transportService.getTaskManager());
+
```

Comment:
If you look at how task manager is wiring up services this injection looks counter-intuitive

Reply:
This was done due to ThreadPool being the first thing being initialised when node is coming up and not having the `TaskManager` reference handy. One option was to make TaskManager singleton explicitly (hold a static reference). But this would cause significant issues to majority of OpenSearch tests which depends on TaskManager because nodes will start sharing same task manager. 

I'll change this to have some lazy loading / Function interface constructs here to avoid having static references but instead rely on some sort of provider which ThreadPool can depend on while node is coming up.